test/test_segment_reductions.py

```diff
@@ -83,21 +85,29 @@ def _test_common(
                 )
             )
 
-    @dtypesIfCUDA(torch.half, torch.bfloat16, torch.float, torch.double)
     @dtypes(torch.half, torch.bfloat16, torch.float, torch.double)
     def test_simple_1d(self, device, dtype):
         lengths = [1, 2, 3, 0]
         data = [1, float("nan"), 3, 4, 5, 5]
-        initial_value = 0
         check_backward = True
 
-        for reduction in ("max", "mean"):
+        for reduction in reductions:
             if reduction == "max":
+                initial_value = 0
                 expected_result = [1, float("nan"), 5, initial_value]
                 expected_grad = [1, 1, 0, 0, 0.5, 0.5]
             elif reduction == "mean":
+                initial_value = 0
                 expected_result = [1, float("nan"), 4.666, initial_value]
                 expected_grad = [1.0, 0.5, 0.5, 0.333, 0.333, 0.333]
+            elif reduction == "min":
+                initial_value = 1000  # some high number
+                expected_result = [1, float("nan"), 4, initial_value]
```

Comment:
instead of creating expected_result as a list, you can create is as a torch.tensor with correct device and dtype, and not convert in the _test_common:
```
expected_result = torch.tensor([1, float("nan"), 4, initial_value], device=device, dtype=dtype)
```